expression/expressions/subquery.go

```diff
@@ -116,3 +110,114 @@ func (sq *SubQuery) Plan(ctx context.Context) (plan.Plan, error) {
 	sq.p, err = sq.Stmt.Plan(ctx)
 	return sq.p, errors.Trace(err)
 }
+
+// EvalRows executes the subquery and returns the multi rows with rowCount.
+// rowCount < 0 means no limit.
+// if the ColumnCount is 1, we will return a column result like {1, 2, 3}
+// otherwise, we will return a table result like {{1, 1}, {2, 2}}
+func (sq *SubQuery) EvalRows(ctx context.Context, args map[interface{}]interface{}, rowCount int) ([]interface{}, error) {
+	p, err := sq.Plan(ctx)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	defer p.Close()
+
+	sq.push(ctx)
+
+	var (
+		row *plan.Row
+		res = []interface{}{}
+	)
+
+	for rowCount != 0 {
+		row, err = p.Next(ctx)
+		if err != nil {
+			break
+		}
+		if row == nil {
+			break
+		}
+		if len(row.Data) == 1 {
```

Comment:
e.g, `select * from t where c1 > any (select c1 from t2)`, we can compare c1 with a row in table like `c1 > {1}`, only support `c1 > 1`.